functions/Invoke-DbaBalanceDataFiles.ps1

```diff
@@ -84,113 +92,109 @@ function Invoke-DbaBalanceDataFiles {
+			try {
+				$Server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential
+			}
+			catch {
+				Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
 			}
 			
-			$DatabaseCollection = $server.Databases | Where-Object { $_.Name -in $Database }
-		}
-		else {
-			Stop-Function -Message "Please supply a database to balance out" -Target $SqlInstance -Continue
-		}
-		
-		# Get the server version
-		$serverVersion = $server.Version.Major
-		
-		# Check edition of the sql instance
-		if ($RebuildOffline) {
-			Write-Message -Message "Continuing with offline rebuild." -Level Verbose
-		}
-		elseif (-not $RebuildOffline -and ($serverVersion -lt 9 -or (([string]$Server.Edition -notlike "Developer*") -and ($Server.Edition -notlike "Enterprise*")))) {
-			# Set up the confirm part
-			$message = "The server does not support online rebuilds of indexes. `nDo you want to rebuild the indexes offline?"
-			$choiceYes = New-Object System.Management.Automation.Host.ChoiceDescription "&Yes", "Answer Yes."
-			$choiceNo = New-Object System.Management.Automation.Host.ChoiceDescription "&No", "Answer No."
-			$options = [System.Management.Automation.Host.ChoiceDescription[]]($choiceYes, $choiceNo)
-			$result = $host.ui.PromptForChoice($title, $message, $options, 0)
-			
-			# Check the result from the confirm
-			switch ($result) {
-				# If yes
-				0 {
-					# Set the option to generate a full backup
-					Write-Message -Message "Continuing with offline rebuild." -Level Verbose
-					
-					[bool]$supportOnlineRebuild = $false
-				}
-				1 {
-					Stop-Function -Message "You chose to not allow offline rebuilds of indexes. Use -RebuildOffline" -Target $SqlInstance
-					return
+			# Check the database parameter
+			if ($Database) {
+				if ($Database -notin $server.Databases.Name) {
+					Stop-Function -Message "One or more databases cannot be found on instance on instance $instance" -Target $instance -Continue
 				}
-			} # switch
-		}
-		elseif ($serverVersion -ge 9 -and (([string]$Server.Edition -like "Developer*") -or ($Server.Edition -like "Enterprise*"))) {
-			[bool]$supportOnlineRebuild = $true
+				
+				$DatabaseCollection += $server.Databases | Where-Object { $_.Name -in $Database }
+			}
+			else {
+				Stop-Function -Message "Please supply a database to balance out" -Target $instance -Continue
+			}
+			
+			# Get the server version
+			$serverVersion = $server.Version.Major
+			
+			# Check edition of the sql instance
+			if ($RebuildOffline) {
+				Write-Message -Message "Continuing with offline rebuild." -Level Verbose
+			}
+			elseif (-not $RebuildOffline -and ($serverVersion -lt 9 -or (([string]$Server.Edition -notmatch "Developer") -and ($Server.Edition -notmatch "Enterprise")))) {
+				# Set up the confirm part
+				$message = "The server does not support online rebuilds of indexes. `nDo you want to rebuild the indexes offline?"
```

Comment:
Do we need to validate that the databases are involved in mirroring or an AG as well?

Doing this to a database in an AG can cause serious issues on the secondaries if the (1) index is large or (2) it is just a large database in general.

Reply:
Be something like "This database is involved in an AG or mirror, are you sure you want to continue?"

Reply:
Check for Log Shipping as well...big time issue in some client servers with the standby server getting behind because of index maintenance.